antarest/storage/service.py

```diff
@@ -397,6 +400,42 @@ def delete_output(
             params.get_user_id(),
         )
 
+    def download_outputs(
+        self,
+        study_id: str,
+        output_id: str,
+        data: StudyDownloadDTO,
+        params: RequestParameters,
+    ) -> MatrixAggregationResult:
+        """
+        Download outputs
+        Args:
+            study_id: study Id
+            output_id: output id
+            data: Json parameters
+            params: request parameters
+
+        Returns: CSV content file
+
+        """
+        # GET STUDY ID
+        study = self._get_study(study_id)
+        self._assert_permission(params.user, study, StudyPermissionType.READ)
+
+        if not isinstance(study, RawStudy):
+            raise StudyTypeUnsupported(study_id, study.type)
+
+        self.logger.info(
+            "study %s data asked by user %s",
+            study_id,
+            params.get_user_id(),
+        )
+        url = f"/output/{output_id}/economy"
```

Comment:
you can declare url inside the "build" method.
also, the last part of the path is not always "economy", it can also be "adequacy" or "adequacy-draft" depending on the simulation mode that you can find in the StudyConfig.
Also, in "ConfigPathBuilder.parse_simulation" and "ConfigJsonBuilder.parse_simulation", it should be added the mode "dft" => "draft".
||Economy|Adequacy|Draft|
|-|-|-|-|
|name|economy|adequacy|draft|
|folder-name|economy|adequacy|adequacy-draft|
|simulation-id-suffix|eco|adq|dft|